source/solvers/gls_navier_stokes.cc

```diff
@@ -867,9 +867,13 @@ GLSNavierStokesSolver<dim>::assemble_local_system_matrix(
 
   copy_data.reset();
 
+  const bool using_immersed_boundary_solid =
+    (this->multiphysics->get_immersed_solid_signed_distance_function()) &&
+    this->simulation_parameters.physical_properties_manager
+        .get_number_of_solids() > 0;
```

Comment:
I'm not sure here, but maybe it should be restricted to one immersed solid

```suggestion
        .get_number_of_solids() == 1;
```